### PR TITLE
updating v2-4 models for phenopackets v2 biosample

### PR DESCRIPTION
**Pull request by mbaudis**
This is a first attempt to bring the biosample in the models in line with phenopackets v2. It adds a "common" directory, since some of the classes there will be re-used for individual etc.

### CODE REVIEW

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json [hunk 1]
```diff
@@ -5,79 +5,164 @@
...
+      "$ref": "https://raw.githubusercontent.com/ga4gh-beacon/beacon-framework-v2/main/common/ontologyTerm.json",
+      "example": {
+        "id": "EFO:0009656",
+        "label": "neoplastic sample"
+      }      
+    },
+    "sampleProcessing": {
+      "description": "Status of how the specimen was processed.",
+      "$ref": "https://raw.githubusercontent.com/ga4gh-beacon/beacon-framework-v2/main/common/ontologyTerm.json",
+      "example": {
```
Comment (jrambla) [inline BEACON-V2-draft4-Model/biosamples/defaultSchema.json:149]: Is the example repeated from "sampleProcessing"?

**Review by jrambla — CHANGES_REQUESTED** [review]
I like the common directory idea, as we have in the Framework. It makes sense.
I'm also in favour of including the Phenopackets schema in parallel to the Beacon schema, like we do with the ga4gh-service-info schema.
I'm far less convinced that the Beacon v2 Model should carbon copy the Phenopackets one. I guess it brings an undesired message that Beacon is relaying in external schemas.

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json [hunk 2]
```diff
@@ -5,79 +5,164 @@
   "description": "Schema for an biosample entry.",
   "type":"object",
   "properties": {
-    "biosampleId": {
```
[the remark this answers is not in the document]
Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:8]: So far in GA4GH schemas have always used `id` for their own id, and `otherSchemaId` to point to it. I.e. a biosample's `id` is always referenced to as `biosampleId`. This has advantages for implementation, in that one does not have to design different options or compose the id parameter name when parsing over different objects (i.e. searching for an object by its `id` value).
Also: see [Phenopackets](https://phenopacket-schema.readthedocs.io/en/v2/biosample.html) etc.
No need to change this behaviour now...

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json [hunk 3]
```diff
@@ -5,79 +5,164 @@
...
+    "id": {
       "description": "Biosample identifier (external accession or internal ID).",
       "type": "string",
-      "example": "S0001"
+      "example": "pgxbs-kftva634"
```
[the remark this answers is not in the document]
Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:11]: Important :-)

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json [hunk 4]
```diff
@@ -5,79 +5,164 @@
...
+      "description": "Class for the type of material, e.g., RNA, DNA, cultured cells.",
+      "$ref": "https://raw.githubusercontent.com/ga4gh-beacon/beacon-framework-v2/main/common/ontologyTerm.json",
+      "example": {
+        "id" : "EFO:0008479",
+        "label" : "genomic DNA"
+      }
+    },
+    "phenotypicFeatures": {
+      "description": "List of phenotypic abnormalities of the sample. RECOMMENDED.",
+      "type": "list",
```
[the remark this answers is not in the document]
Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:46]: Fixing ...

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json [hunk 5]
```diff
@@ -5,79 +5,164 @@
...
-      "example": "S0001"
+      "example": "pgxbs-kftva634"
     },
     "individualId": {
       "description": "Reference to individual ID (Individual.individualId).",
       "type": "string",
-      "example": "P0001"
+      "example": "pgxind-kftx26aj"
+    },
+    "derivedFromId": {
```
[the remark this answers is not in the document]
Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:18]: +1 for getting rid of this, too; I'm not a fan either. I followed the discussion; it was mostly about a vagueness of concept was a biosample is, some conflating it w/ material / biomolecule type and then just leaving open the option to do "biosample which is tissue" => "biosample which is the DNA ... derived from this tissue".

 Will change.

File: BEACON-V2-draft4-Model/biosamples/defaultSchema.json (hunk 1 above)
Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:149]: C&P from "materialSample" ... :-/

Comment (mbaudis) [reply BEACON-V2-draft4-Model/biosamples/defaultSchema.json:149]: There was no example in the PXF documentation. I understand what this is getting at & now provide an example, but am also fine w/ having it removed (`sampleProcessing` and `sampleStorage`).

**Review by jrambla — APPROVED** [review]

**Pull request merged**